helix-view/src/editor.rs

```diff
@@ -1251,15 +1249,18 @@ impl Editor {
 
     fn replace_document_in_view(&mut self, current_view: ViewId, doc_id: DocumentId) {
         let view = self.tree.get_mut(current_view);
-        view.doc = doc_id;
-        view.offset = ViewPosition::default();
 
+        // If swapping from a document, record the last view position
+        if let Some(old_doc) = self.documents.get_mut(&view.doc) {
+            old_doc.last_view_position = view.offset;
+        }
+
+        view.doc = doc_id;
         let doc = doc_mut!(self, &doc_id);
+        view.offset = doc.last_view_position;
         doc.ensure_view_init(view.id);
         view.sync_changes(doc);
         doc.mark_as_focused();
-
-        align_view(doc, view, Align::Center);
```

Comment:
The align_view call here was previously ensuring that the cursor was in-view and we will still need to do that with this change:

```rust
// at the beginning of the function:
let scrolloff = self.config().scrolloff;
// ...
// at the end:
view.ensure_cursor_in_view(doc, scrolloff);
```

Updating the `view.offset` will adjust the scroll and that may leave the cursor outside the visible area. See https://asciinema.org/a/Uxh9FvcWFi2B26yYE2yb6Kfun: I leave the old view position of `README.md` at the end of the buffer, create a new split View and focus `README.md`. The scroll of the buffer is updated to the last view position but the cursor is at the beginning, so any movements (hjkl) or commands will move the viewport to the cursor.